### PR TITLE
Support for Mzaalo

### PR DESCRIPTION
**Pull request by ItzMaxTV**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Mzaalo, a site for movies, shows, and m-play (clips) with support of multiple languages such as Bengali, Hindi, Kannada, etc..

```
[debug] Loaded 1814 extractors 
[Mzaalo] Extracting URL: https://www.mzaalo.com/play/movie/c0958d9f-f90e-4503-a755-44358758921d/Jamun 
[Mzaalo] c0958d9f-f90e-4503-a755-44358758921d: Downloading JSON metadata 
[Mzaalo] c0958d9f-f90e-4503-a755-44358758921d: Downloading m3u8 information 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id 
[info] c0958d9f-f90e-4503-a755-44358758921d: Downloading 1 format(s): 3661
[info] Writing video metadata as JSON to: test_Mzaalo_c0958d9f-f90e-4503-a755-44358758921d.info.json 
[debug] Invoking hlsnative downloader on "https://content-aaps1-ur-fifa1.uplynk.com/636ad16a1c554df0a0958716fab45902/f.m3u8?tc=1&exp=1685320198&rn=2074360713&ct=a&eid=7005875&oid=12ccd788e5224b0692264ac02cc4929f&ddp=1&rates=&sig=5d991bfbcfddc8b79ea17a6d066429307a692631297dd5586f72e85070b80c59&pbs=a17377135b804af29d45088061809265"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 1350 
[download] Destination: test_Mzaalo_c0958d9f-f90e-4503-a755-44358758921d.mp4
[download] 100% of  752.56KiB in 00:00:00 at 1.14MiB/s
[debug] Loaded 1814 extractors 
[Mzaalo] Extracting URL: https://www.mzaalo.com/play/original/93d42b2b-f373-4c2d-bca4-997412cb069d/Modi-Season-2-CM-TO-PM/Episode-1:Decision,-Not-Promises 
[Mzaalo] 93d42b2b-f373-4c2d-bca4-997412cb069d: Downloading JSON metadata
[Mzaalo] 93d42b2b-f373-4c2d-bca4-997412cb069d: Downloading m3u8 information 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id 
[info] 93d42b2b-f373-4c2d-bca4-997412cb069d: Downloading 1 format(s): 3712 
[info] Writing video metadata as JSON to: test_Mzaalo_1_93d42b2b-f373-4c2d-bca4-997412cb069d.info.json
[debug] Invoking hlsnative downloader on "https://content-aaps1-ur-fifa1.uplynk.com/50c3ca8e3e014ab4a4991f80e3481298/f.m3u8?tc=1&exp=1685320205&rn=1677183904&ct=a&eid=7004585&oid=12ccd788e5224b0692264ac02cc4929f&ddp=1&rates=&sig=2c214532de04b046c72fe4b327a65b6ddf60a5e104176d26f83c7c08a65ffeb8&pbs=c910ba8ed6c849548a912627d210ef15"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 499 
[download] Destination: test_Mzaalo_1_93d42b2b-f373-4c2d-bca4-997412cb069d.mp4 
[download] 100% of    1.13MiB in 00:00:00 at 2.51MiB/s
[debug] Loaded 1814 extractors 
[Mzaalo] Extracting URL: https://www.mzaalo.com/play/clip/83cdbcb5-400a-42f1-a1d2-459053cfbda5/Manto-Ki-Kahaaniya 
[Mzaalo] 83cdbcb5-400a-42f1-a1d2-459053cfbda5: Downloading JSON metadata
[Mzaalo] 83cdbcb5-400a-42f1-a1d2-459053cfbda5: Downloading m3u8 information 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id 
[info] 83cdbcb5-400a-42f1-a1d2-459053cfbda5: Downloading 1 format(s): 3828 
[info] Writing video metadata as JSON to: test_Mzaalo_2_83cdbcb5-400a-42f1-a1d2-459053cfbda5.info.json 
[debug] Invoking hlsnative downloader on "https://mzaalovod.pc.cdn.bitgravity.com/inverleigh-output/Manto_ki_kahaaniya_film/3700k/HLS/Manto_ki_kahaaniya_film_3700k.m3u8"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 194 
[download] Destination: test_Mzaalo_2_83cdbcb5-400a-42f1-a1d2-459053cfbda5.mp4 
[download] 100% of    6.33MiB in 00:00:13 at 483.52KiB/s
ok

----------------------------------------------------------------------
Ran 1 test in 27.727s

OK
```

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🎥🌐🆕

This pull request adds support for downloading videos from Mzaalo and Weyyak, two streaming platforms for Indian and Arabic content. It introduces two new modules, `mzaalo.py` and `weyyak.py`, which define the extractor classes for these platforms. The classes implement the logic for extracting the video metadata and URLs from the JSON data.

> _We're the brave and bold extractors of the video streams_
> _We parse the URLs and download the JSON schemes_
> _We fetch the metadata and formats from `MzaaloIE` and `WeyyakIE`_
> _And we return the dictionaries for `youtube-dl` to see_

### Walkthrough
*  Add support for extracting videos from Mzaalo and Weyyak platforms ([link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R1149), [link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R2319), [link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-fca616a626bd38f07a1c31fea748732ba572ba03f3cfd6526cf54d5a6919314fR1-R89), [link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-6b68740474cd9fe80bd83d02321608922d08b15640802018b9986f83e79dd53dR1-R50))
  * Import `MzaaloIE` and `WeyyakIE` classes from `mzaalo.py` and `weyyak.py` modules respectively in `_extractors.py` ([link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R1149), [link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R2319))
  * Define `MzaaloIE` class in `mzaalo.py` that inherits from `InfoExtractor` and implements `_real_extract` method ([link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-fca616a626bd38f07a1c31fea748732ba572ba03f3cfd6526cf54d5a6919314fR1-R89))
  * Define `WeyyakIE` class in `weyyak.py` that inherits from `InfoExtractor` and implements `_real_extract` method ([link](https://github.com/yt-dlp/yt-dlp/pull/7163/files?diff=unified&w=0#diff-6b68740474cd9fe80bd83d02321608922d08b15640802018b9986f83e79dd53dR1-R50))

</details>

### CODE REVIEW

File: yt_dlp/extractor/mzaalo.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+from ..utils import (
+    parse_age_limit,
+    parse_duration,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class MzaaloIE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/www\.mzaalo\.com\/play\/(?P<type>movie|original|clip)\/(?P<id>[a-fA-F0-9-]+)\/[a-zA-Z0-9_-]+'
```
Comment (pukkandan) [inline yt_dlp/extractor/mzaalo.py:11]: * Unnecessary escape for `/`
* `[a-zA-Z0-9_-]` => `[\w-]

File: yt_dlp/extractor/mzaalo.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'duration': 1937.0,
+            'language': 'Hin',
+            'asset_type': 'CLIP',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id, type_ = self._match_valid_url(url).group('id', 'type')
+        path = f'partner/streamurl?&assetId={video_id}&getClipDetails=YES' if type_ == 'clip' else f'api/v2/player/details?assetType={type_.upper()}&assetId={video_id}'
```
Comment (pukkandan) [inline yt_dlp/extractor/mzaalo.py:67]:
```suggestion
        path = (f'partner/streamurl?&assetId={video_id}&getClipDetails=YES' if type_ == 'clip'
                else f'api/v2/player/details?assetType={type_.upper()}&assetId={video_id}')
```
line too long

File: yt_dlp/extractor/weyyak.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import re
```
Comment (pukkandan) [inline yt_dlp/extractor/weyyak.py:1]: Why's this file in this PR?

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/weyyak.py (hunk 3 above)
Comment (ItzMaxTV) [reply yt_dlp/extractor/weyyak.py:1]: This was the other PR, it somehow linked to this..

File: yt_dlp/extractor/mzaalo.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/mzaalo.py:11]: This was not adressed

File: yt_dlp/extractor/mzaalo.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    def _real_extract(self, url):
+        video_id, type_ = self._match_valid_url(url).group('id', 'type')
+        path = (f'partner/streamurl?&assetId={video_id}&getClipDetails=YES' if type_ == 'clip'
+                else f'api/v2/player/details?assetType={type_.upper()}&assetId={video_id}')
+        data = self._download_json(
+            f'https://production.mzaalo.com/platform/{path}', video_id, headers={
+                'Ocp-Apim-Subscription-Key': '1d0caac2702049b89a305929fdf4cbae',
+            })['data']
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(data["streamURL"], video_id)
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:73]: single-quote
```suggestion
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(data['streamURL'], video_id)
```
&nbsp;
Also, it looks like there aren't any subtitles to be extracted from the m3u8, but there are webvtt subtitles in the API JSON / `data` var, e.g.:
```json
    "subtitles": {
      "ara": "https://subtitles-a.erosnow.com/full_length/7/1069317/7005875/4000453-ara.webvtt",
      "eng": "https://subtitles-a.erosnow.com/full_length/7/1069317/7005875/7006053-eng.webvtt"
    },
```

File: yt_dlp/extractor/mzaalo.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+from ..utils import (
+    parse_age_limit,
+    parse_duration,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class MzaaloIE(InfoExtractor):
+    _VALID_URL = r'https?://www.mzaalo.com/play/(?P<type>movie|original|clip)/(?P<id>[a-fA-F0-9-]+)/[\w-]+'
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:11]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?mzaalo\.com/play/(?P<type>movie|original|clip)/(?P<id>[a-fA-F0-9-]+)/[\w-]+'
```
we do want to escape dots in hostnames, and make the subdomain optional

File: yt_dlp/extractor/mzaalo.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            'subtitles': subtitles,
+            **traverse_obj(data, {
+                'title': ('title', {str}),
+                'description': ('description', {str}),
+                'duration': ('duration', {parse_duration}),
+                'age_limit': ('maturity_rating', {parse_age_limit}),
+                'thumbnails': ('images', ..., {'url': {url_or_none}}),
+                'categories': ('genre', ..., {str}),
+                'language': ('language', {str}),
+                'asset_type': ('asset_type', {str}),
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:93]:
```suggestion
```
this is not a common field

File: yt_dlp/extractor/mzaalo.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            'id': 'c0958d9f-f90e-4503-a755-44358758921d',
+            'title': 'Jamun',
+            'ext': 'mp4',
+            'description': 'md5:24fe9ebb9bbe5b36f7b54b90ab1e2f31',
+            'thumbnails': 'count:15',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'duration': 5527.0,
+            'language': 'hin',
+            'categories': ['Drama'],
+            'asset_type': 'MOVIE',
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:26]:
```suggestion
```

File: yt_dlp/extractor/mzaalo.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            'id': '93d42b2b-f373-4c2d-bca4-997412cb069d',
+            'title': 'Episode 1:Decision, Not Promises',
+            'ext': 'mp4',
+            'description': 'md5:16f76058432a54774fbb2561a1955652',
+            'thumbnails': 'count:22',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'duration': 2040.0,
+            'language': 'hin',
+            'categories': ['Drama'],
+            'asset_type': 'ORIGINAL',
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:44]:
```suggestion
```

File: yt_dlp/extractor/mzaalo.py [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        'info_dict': {
+            'id': '83cdbcb5-400a-42f1-a1d2-459053cfbda5',
+            'title': 'Manto Ki Kahaaniya',
+            'ext': 'mp4',
+            'description': 'md5:c3c5f1d05f0fd1bfcb05b673d1cc9f2f',
+            'thumbnails': 'count:3',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'duration': 1937.0,
+            'language': 'Hin',
+            'asset_type': 'CLIP',
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:61]:
```suggestion
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/mzaalo.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+            'title': 'Jamun',
+            'ext': 'mp4',
+            'description': 'md5:24fe9ebb9bbe5b36f7b54b90ab1e2f31',
+            'thumbnails': 'count:15',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'duration': 5527.0,
+            'language': 'hin',
+            'categories': ['Drama'],
+            'age_limit': 13,
+            'content_discriptor': 'Alcohol',
```
Comment (bashonly) [inline yt_dlp/extractor/mzaalo.py:27]:
```suggestion
```

File: yt_dlp/extractor/mzaalo.py [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            'formats': formats,
+            'subtitles': subtitles,
+            **traverse_obj(data, {
+                'title': ('title', {str}),
+                'description': ('description', {str}),
+                'duration': ('duration', {parse_duration}),
+                'age_limit': ('maturity_rating', {parse_age_limit}),
+                'thumbnails': ('images', ..., {'url': {url_or_none}}),
+                'categories': ('genre', ..., {str}),
+                'language': ('language', {str}),
```
Comment (pukkandan) [inline yt_dlp/extractor/mzaalo.py:87]: language should be set for each applicable format, not for the video as a whole.

**Pull request merged**